internal/consumer/service.go

```diff
@@ -0,0 +1,81 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+
+	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type Service struct {
+	logger          *zap.Logger
+	numberOfWorkers int
+	hnClient        hackernews.Client
+}
+
+type Client interface {
+	processStories(ctx context.Context)
+}
+
+func NewService(logger *zap.Logger, config *model.ConsumerConfig, hnClient hackernews.Client) (*Service, error) {
+	if hnClient == nil {
+		var err error
+		hnClient, err = hackernews.New(config.BaseUrl, nil)
```

Comment:
This could be changed to an assignment (`:=`) to avoid declaring the `var err error` above

Reply:
Changing the assignment means I cannot reuse the hnClient variable, hence why I implemented it like this

Reply:
Apologies, I forgot `:=` would try and reassign `hnClient` because it's in the `if` scope.